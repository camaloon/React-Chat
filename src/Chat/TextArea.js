import React from 'react';

export default () => <textarea id="" name="" cols="30" rows="10"></textarea>
