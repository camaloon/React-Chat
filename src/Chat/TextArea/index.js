import React, { Component } from 'react';
import store from '../../lib/state';
import { addMessage } from '../../lib/reducers/messageReducer';


class TextArea extends Component {
    constructor(props) {
        super(props);
        this.state = { text: '' };
        this.onInput = this.onInput.bind(this);
        this.onKeyDown = this.onKeyDown.bind(this);
    }

    onInput(event) {
        this.setState({ text: event.target.value });
    }

    onKeyDown(event) {
        if(event.keyCode === 13){
            store.dispatch(addMessage({ id: Date.now(), room_id: 1, body: event.target.value, from_id: 1 }));
            this.setState({ text: '' })
        }
    }

    render() {
        return <textarea id="" name="" cols="30" rows="10" value={this.state.text} onInput={this.onInput} onKeyDown={this.onKeyDown}></textarea>
    }
}

export default TextArea;
