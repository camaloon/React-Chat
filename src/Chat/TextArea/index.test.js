import React from 'react';
import TextArea from './index';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <TextArea/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('value changes when input is modified', () => {
    // const component = renderer.create(<TextArea/>);
    // let tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
    //
    // // manually trigger the callback
    // tree.props.onMouseEnter();
    // // re-rendering
    // tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
    //
    // // manually trigger the callback
    // tree.props.onMouseLeave();
    // // re-rendering
    // tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
});