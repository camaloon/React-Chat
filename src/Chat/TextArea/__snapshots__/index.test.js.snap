// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<textarea
  cols="30"
  id=""
  name=""
  onInput={[Function]}
  onKeyDown={[Function]}
  rows="10"
  value=""
/>
`;
